File: src/commonMain/kotlin/com/scoreo/application/ArchiveGameTypeUseCase.kt
package com.scoreo.application

import com.scoreo.domain.port.GameTypeRepository

class ArchiveGameTypeUseCase(private val gameTypeRepository: GameTypeRepository) {
    operator fun invoke(gameTypeId: String) {
        val gameType = gameTypeRepository.findById(gameTypeId)
            ?: throw IllegalArgumentException("Game type not found: $gameTypeId")
        
        val archived = gameType.copy(active = false)
        gameTypeRepository.save(archived)
    }
}

File: apps/scoreo/src/application/archiveGameTypeUseCase.ts
import { NotFoundError } from '../domain/model/errors'
import type { GameTypeRepository } from '../domain/port/gameTypeRepository'

export class ArchiveGameTypeUseCase {
  constructor(private readonly gameTypeRepository: GameTypeRepository) {}

  invoke(gameTypeId: string): void {
    const gameType = this.gameTypeRepository.findById(gameTypeId)
    if (!gameType) throw new NotFoundError('GameType', gameTypeId)

    this.gameTypeRepository.save({ ...gameType, active: false })
  }
}
